### PR TITLE
Getting benchmark.py working again and migrating the wiki

### PR DESCRIPTION
**Pull request by DanHickstein**
I am taking all of the descriptions of the transform methods out of the wiki and putting them into doc/transform_methods. I think that it looks really good as part of the documentation. 

Also, I made benchmark.py sort-of work again. It still needs some more work, and I'm done working on it for not. If @stggh or @DhrubajyotiDas want to take a look at it, that would be great. Currently, I think that it benchmarks the basis set generations as well as the transform methods 5 times each. This is probably too many times for the basis set generation, since this takes a long time anyway.

### CODE REVIEW

**Comment by DanHickstein** [comment]
Phew, I finally finished migrating all of the wiki over to the actual documentation. All of the files now live in `docs/transform_methods`. I changed some things, added some things, and may have messed up your carefully worded docs in the process. 

@DhrubajyotiDas, the example in the three-point doc doesn't work any more. I got it to produce a plot, but it doesn't seem to be accurate. Please have a look and see if you'd like to spruce it up or get rid of it.

Have a look at how nice these docs look! 
http://pyabel-dh-fork.readthedocs.org/en/latest/transform_methods.html

I also included a placeholder for a section where we can put some various benchmarks of the speed and accuracy of the transforms (which I assume should be lots of fun):
http://pyabel-dh-fork.readthedocs.org/en/latest/transform_methods/comparison.html

File: doc/transform_methods/basex.rst [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+
+How to use it
+-------------
+
+The recommended way to complete the inverse Abel transform using the BASEX algorithm for a full image is to use the :func:`abel.transform` function: ::
+
+	abel.transform(myImage, method='basex', direction='inverse')
+
+Note that the forwrad BASEX transform is not yet implemented in PyAbel.
```
Comment (DhrubajyotiDas) [inline doc/transform_methods/basex.rst]: Typo here in `forward`.

**Comment by stggh** [comment]
Beautiful. @DanHickstein you are the doc king.

Time for me to fix some of my descriptions.

BTW, I think that you are a little harsh on the Onion-peeling method. It is pretty close. There is possibly only a small bug-fix required?

**Comment by DhrubajyotiDas** [comment]
The three_point example is broken because the raw data being fed needs to be cut in half (like the BASEX step example). I'll take care of it tomorrow (in about 12 hours) if you haven't already tidied it up by then.

**Comment by stggh** [comment]
> I made benchmark.py sort-of work again

Excellent work. The default call works. 
Note, `hansenlaw` also has a `forward` method.
```
etalon:.+-DH/PyAbel/abel]$ diff benchmark.py ~/Downloads/benchmark-old.py 
37c37
<         res_fabel = {'HansenLaw': {'tr': []}}
---
>         res_fabel = {}
70,72d69
<                     number=NREPEAT)/NREPEAT)
<             res_fabel['HansenLaw']['tr'].append(
<                 Timer(lambda: abel.hansenlaw.hansenlaw_transform(x, direction='forward')).timeit(
```

```
In [1]: import abel

In [2]: abel.benchmark.AbelTiming()
Out[2]: 
PyAbel benchmark run on x86_64

 =========     Direct Abel implementations ==========
time to solution [s] -> transform                         (basis sets generation)

 Implementation |    n = 201          |    n = 401          
 ----------------------------------------------------------
  direct_Python |  1.8e-02                                 |  1.1e-01                                 
      HansenLaw |  1.3e-02                                 |  3.2e-02                                 

 =========    Inverse Abel implementations ==========
time to solution [s] -> transform                         (basis sets generation)

 Implementation |    n = 201          |    n = 401          
 ----------------------------------------------------------
          BASEX |  1.2e-02                                 ( 1.1e-03) |  1.0e-01                                 ( 2.8e-03) 
    Three_point |  8.0e-03                                 ( 4.0e-04) |  8.0e-03                                 ( 5.0e-04) 
      HansenLaw |  1.4e-02                                 |  3.6e-02                                 
  direct_Python |  2.5e-02                                 |  1.8e-01                                 

```

**Comment by DanHickstein** [comment]
I fixed the typo in basex.rst (thanks @DhrubajyotiDas!)

I probably won't have time to work on this much tomorrow, so feel free to merge this and make additional changes to the docs, examples, and benchmark.py.

**Comment by stggh** [comment]
Ok, good work today!

benchmark run directly:
```
[etalon:.+yAbel-DH/PyAbel]$ python abel/benchmark.py
PyAbel benchmark run on x86_64

 =========     Direct Abel implementations ==========
time to solution [s] -> transform                         (basis sets generation)

 Implementation |    n = 501          
 ----------------------------------------
  direct_Python |  2.2e-01                                 
      HansenLaw |  4.2e-02                                 
       direct_C |  7.0e-02                                 

 =========    Inverse Abel implementations ==========
time to solution [s] -> transform                         (basis sets generation)

 Implementation |    n = 501          
 ----------------------------------------
          BASEX |  2.5e-01                                 ( 2.6e-02) 
    Three_point |  1.2e-01                                 ( 1.8e+00) 
       direct_C |  1.9e-01                                 
      HansenLaw |  4.7e-02                                 
  direct_Python |  3.5e-01   
```

**Comment by DanHickstein** [comment]
Ok, I'm going to merge this now. 

@stgg, regarding Onion Peeling - I agree, it is pretty close. I felt like it was doing a good job on spherically symmetric datasets, but on something like a cylindrical gaussian or a step function, it was failing. I don't think that it's supposed to require symmetry higher than cylindrical, but this might be my misunderstanding of the algorithm. If you have a chance to work on this, I would appreciate it.

**Comment by DhrubajyotiDas** [comment]
@DanHickstein I've taken care of the three_point example with the latest PR #127. If you could verify that it builds properly, then you can go ahead and merge it.

**Pull request merged**